### PR TITLE
Handle provisioning of device if it’s not in managed state for longer…

### PR DESCRIPTION
**Pull request by Abhishek-121**
… time, make output msg more readable, device credential update without giving each parameter etc.

In this commit -

— Handle the updation of device credentials without giving all required parameter including the updation of management Ip Address.

— Write code for wired device provisioning if devices not coming to managed for longer period of time and move to provisioning for other device.

— Make return self.msg more readable and clear after each operation.

— Handle the failure message when importing some junk image.

— Handle the case when input IP is dump making it to give message and return without running the resync operation.

— Make the message for reboot device clear in case of failure. 
 
-- Add backward computability of variable name in site module.

-- Add the log with critical level  along with device IP for which swim distribution/activation gets failed in partial success scenario

### CODE REVIEW

File: plugins/modules/inventory_intent.py [hunk 1]
```diff
@@ -1296,6 +1303,8 @@ def reboot_access_points(self):
                     self.status = "success"
                     self.result['changed'] = True
                     self.result['response'] = execution_details
+                    self.log("AP Devices Rebooted Successfully and Rebooted devices are :" + str(device_ips))
+                    self.msg = "Device " + str(device_ips) + " Rebooted Successfully !!"
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:1307]: self.msg = "Device(s) {} successfully rebooted!".format(str(device_ips))

File: plugins/modules/inventory_intent.py [hunk 2]
```diff
@@ -2213,13 +2236,54 @@ def get_diff_merged(self, config):
...
+                    'enable_password': 'enablePassword',
+                    'snmp_username': 'snmpUserName'
+                }
+                device_update_key_list = ["username", "password", "enable_password", "cli_transport", "snmp_username"]
+
+                for key in device_update_key_list:
+                    mapped_key = device_key_mapping[key]
+
+                    if playbook_params[mapped_key] is None:
+
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:2275]: removed.

File: plugins/modules/inventory_intent.py [hunk 3]
```diff
@@ -2213,13 +2236,54 @@ def get_diff_merged(self, config):
 
             if credential_update:
                 # Update Device details and credentails
+                device_uuids = self.get_device_ids(device_to_update)
+                password = "Testing@123"
+                payload_params = {"deviceUuids": device_uuids, "password": password, "operationEnum": "0"}
+                response = self.trigger_export_api(payload_params)
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:2242]: changed name to export_response

File: plugins/modules/inventory_intent.py [hunk 4]
```diff
@@ -2213,13 +2236,54 @@ def get_diff_merged(self, config):
 
             if credential_update:
                 # Update Device details and credentails
+                device_uuids = self.get_device_ids(device_to_update)
+                password = "Testing@123"
+                payload_params = {"deviceUuids": device_uuids, "password": password, "operationEnum": "0"}
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:2241]: added the new name.

File: plugins/modules/swim_intent.py [hunk 5]
```diff
@@ -910,9 +913,10 @@ def get_diff_import(self):
             return self
 
         except Exception as e:
-            self.log("Import Image details are not given in the playbook")
             self.status = "failed"
-            self.result['changed'] = False
+            self.msg = "Import Image details are not given in the playbook or Import Image API not triggered successfully."
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:917]: Added.

File: plugins/modules/inventory_intent.py [hunk 6]
```diff
@@ -1485,6 +1493,14 @@ def provisioned_wired_device(self):
                         and response.get("hostname")
                     ):
                         break
+                    count = count + 1
+                    if count > 200:
+                        managed_flag = False
+                        break
+
+                if not managed_flag:
+                    self.log("Device {0} not coming to managed state so cannot perform provisioning operation".format(device_ip))
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:1502]: We are not continuing in while loop, it's in for loop to move to next device.
In While loop we waited for device to come in managed state if it not come for a certain period of time we moved to next device logging the message.

File: plugins/modules/inventory_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:1307]: added.

File: plugins/modules/inventory_intent.py [hunk 7]
```diff
@@ -1296,6 +1303,8 @@ def reboot_access_points(self):
                     self.status = "success"
                     self.result['changed'] = True
                     self.result['response'] = execution_details
+                    self.log("AP Devices Rebooted Successfully and Rebooted devices are :" + str(device_ips))
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:1306]: added.

File: plugins/modules/inventory_intent.py [hunk 8]
```diff
@@ -1174,6 +1173,14 @@ def resync_devices(self):
                 device_ips.remove(ap_ip)
             self.log("Following devices {0} are AP, so can't perform resync operation.".format(str(ap_devices)))
 
+        if not device_ips:
+            self.msg = "Cannot perform the Resync operation as device's {0} are not present in Cisco Catalyst Center".format(str(device_ips))
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:1177]: changed the log message.

File: plugins/modules/inventory_intent.py [hunk 9]
```diff
@@ -28,6 +28,10 @@
     description: Set to True to verify the Cisco DNA Center config after applying the playbook config.
     type: bool
     default: False
+  dnac_log_level:
+    description: Log levels are used to categorize the logs based on their severity.
+    type: str
+    default: INFO
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:34]: How the customers know other options?
Can we specify the choices?
    choices: [ CRITICAL, ... INFO,... ]

File: plugins/modules/inventory_intent.py [hunk 10]
```diff
@@ -28,6 +28,10 @@
     description: Set to True to verify the Cisco DNA Center config after applying the playbook config.
     type: bool
     default: False
+  dnac_log_level:
+    description: Log levels are used to categorize the logs based on their severity.
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:32]: description: Defines log levels to categorize logs by severity.

File: plugins/modules/swim_intent.py [hunk 11]
```diff
@@ -30,6 +30,10 @@
         Rishita Chowdhary (@rishitachowdhary)
         Abhishek Maheshwari (@abmahesh)
 options:
+  dnac_log_level:
+    description: Log levels are used to categorize the logs based on their severity.
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:34]: same as above

File: plugins/modules/inventory_intent.py (hunk 9 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:34]: added.

File: plugins/modules/inventory_intent.py (hunk 10 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:32]: added.

File: plugins/modules/swim_intent.py (hunk 11 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:34]: added.

File: plugins/modules/inventory_intent.py [hunk 12]
```diff
@@ -28,6 +28,11 @@
     description: Set to True to verify the Cisco DNA Center config after applying the playbook config.
     type: bool
     default: False
+  dnac_log_level:
+    description: Defines log levels to categorize logs by severity.
+        Choices - [DEBUG, , INFO, WARNING, CRITICAL, ERROR]
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:33]: Should be in order, start from CRITICAL

Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py]: added the options in order of their severity.

File: plugins/modules/inventory_intent.py [hunk 13]
```diff
@@ -28,6 +28,11 @@
     description: Set to True to verify the Cisco DNA Center config after applying the playbook config.
     type: bool
     default: False
+  dnac_log_level:
+    description: Defines log levels to categorize logs by severity.
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:32]: description: Specifies the log level for Cisco Catalyst Center logging, categorizing logs by severity.

File: plugins/modules/swim_intent.py [hunk 14]
```diff
@@ -30,6 +30,11 @@
         Rishita Chowdhary (@rishitachowdhary)
         Abhishek Maheshwari (@abmahesh)
 options:
+  dnac_log_level:
+    description: Defines log levels to categorize logs by severity.
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:34]: description: Specifies the log level for Cisco Catalyst Center logging, categorizing logs by severity.

File: plugins/modules/swim_intent.py [hunk 15]
```diff
@@ -809,6 +819,15 @@ def get_diff_import(self):
 
         try:
             import_type = self.want.get("import_type")
+
+            if not import_type:
+                self.msg = "For Importig SWIM Image, details are not provided"
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:824]: self.msg = "Error: Details required for importing SWIM image. Please provide the necessary information."

File: plugins/modules/swim_intent.py [hunk 16]
```diff
@@ -910,9 +932,10 @@ def get_diff_import(self):
             return self
 
         except Exception as e:
-            self.log("Import Image details are not given in the playbook")
             self.status = "failed"
-            self.result['changed'] = False
+            self.msg = "Import Image details are not provided in the playbook, or the Import Image API was not triggered successfully."
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:936]: self.msg = "Error: Import image details are not provided in the playbook, or the Import Image API was not triggered successfully. Please ensure the necessary details are provided and verify the status of the Import Image process."

File: plugins/modules/inventory_intent.py (hunk 13 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:32]: updated.

File: plugins/modules/swim_intent.py (hunk 15 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:824]: added.

File: plugins/modules/swim_intent.py (hunk 14 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:34]: added.

File: plugins/modules/inventory_intent.py [hunk 17]
```diff
@@ -28,6 +28,11 @@
...
+        Options- [CRITICAL, ERROR, WARNING, DEBUG, INFO]
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:33]: Options- [CRITICAL, ERROR, WARNING, INFO, DEBUG]

Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py]: updated.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**